docs/basic-usage.md

```diff
@@ -227,9 +231,15 @@ Even if you develop alone, in six months when reinstalling the project you can f
 the dependencies installed are still working even if your dependencies released many new versions since then.
 (See note below about using the update command.)
 
-{{% note %}}
-For libraries it is not necessary to commit the lock file.
-{{% /note %}}
+#### As a library developer
+
+Library developers have more to consider. Your users are application developers, and your library will run in a Python environment you don't control.
+
+The application ignores your library's lock file. It can use whatever dependency version meets the constraints in your `pyproject.toml`. The application will probably use the latest compatible dependency version. If your library's `poetry.lock` falls behind some new dependency version that breaks things for your users, you're likely to be the last to find out about it.
+
+To work with your library as users are likely to, omit the `poetry.lock` file.
+
+If you still want the reproducibility benefits, consider a regular refresh of `poetry.lock` to stay up-to-date and reduce the risk of sudden breakage for users.
```

Comment:
Another benefit of committing the lock file is that your CI will run faster. (If you have many dependencies locking can take a while.)